### PR TITLE
Improve and Simplify SpinBox Focus Behaviour

### PR DESCRIPTION
**Pull request by nathanfranke**
SpinBox should now respect `focus_mode` correctly.

### CODE REVIEW

**Comment by nathanfranke** [comment]
This seems to work fine for me, but I want at least one other person to test this before we merge. Probably shouldn't cherry pick for 3.2.3, let's wait till 3.2.4

File: scene/gui/spin_box.cpp [hunk 1]
```diff
@@ -261,7 +256,7 @@ SpinBox::SpinBox() {
 	line_edit = memnew(LineEdit);
 	add_child(line_edit);
 
-	line_edit->set_anchors_and_margins_preset(Control::PRESET_WIDE);
+	line_edit->set_anchors_and_margins_preset(PRESET_WIDE);
```
Comment (aaronfranke) [inline scene/gui/spin_box.cpp:259]: This change is questionable, and in any case doesn't really belong in this PR.

**Review by aaronfranke — COMMENTED** [review]
Can you give an example of what the current incorrect behavior is, and how this helps? Also, this needs a rebase.

**Comment by nathanfranke** [comment]
Spinbox focus seems to work fine in 3.4.stable. I can't get any focus to work in master (Just tested project with two buttons).

Assuming fixed already
